tests/std/tests/Dev10_860421_deque_push_back_pop_front/test.cpp

```diff
@@ -259,11 +259,36 @@ void test_exception_safety_for_throwing_movable() {
     assert(d == d_orig);
 }
 
+// Also test GH-4072: shrink_to_fit() should have move_if_noexcept() logic bug
+void test_gh_4072() {
+    constexpr int removed_count = 768;
+
+    deque<ThrowingMovable> d;
+    for (int i = 0; i < 1729; ++i) {
+        d.emplace_back(i);
+    }
+
+    for (int i = 0; i < removed_count; ++i) {
+        d.pop_front();
+        d.pop_back();
+    }
+
+    deque<ThrowingMovable> d2;
+    for (int i = removed_count; i < 1729 - removed_count; ++i) {
+        d2.emplace_back(i);
+    }
+
+    d.shrink_to_fit(); // additionally ensures that no constructor or assignment operator of the element type is called
+    assert(d == d2);
+}
+

```

Comment:
The problem in [the above comment](https://github.com/microsoft/STL/pull/4091/files#r1359805829), we'd better might be able to get caught by:
deque<...> with exactly 128 elements.
pop-front 120 elements.
push-back 1 elements.
shrink_to_fit.